src/common/utils/email/mail.go

```diff
@@ -117,24 +118,77 @@ func sendMailWithTLS(m Mail, auth smtp.Auth, content []byte) error {
 	return client.Quit()
 }
 
-/*
-func loadConfig() {
-	config, err := beego.AppConfig.GetSection("mail")
+// Ping tests the connection and authentication with email server
+// If tls is true, a secure connection is established, or the
+// connection is insecure, and if starttls is true, Ping trys to
+// upgrate the insecure connection to a secure one if email server
+// supports it.
+// Ping doesn't verify the server's certificate and hostname when
+// needed if the parameter insecure is ture
+func Ping(addr, identity, username, password string,
+	timeout int, tls, starttls, insecure bool) (err error) {
+	log.Debugf("establishing TCP connection with %s ...", addr)
+	conn, err := net.DialTimeout("tcp", addr,
+		time.Duration(timeout)*time.Second)
+	if err != nil {
+		return
+	}
+	defer conn.Close()
+
+	host, _, err := net.SplitHostPort(addr)
+	if err != nil {
+		return
+	}
+
+	if tls {
+		log.Debugf("establishing SSL/TLS connection with %s ...", addr)
+		tlsConn := tlspkg.Client(conn, &tlspkg.Config{
+			ServerName:         host,
+			InsecureSkipVerify: insecure,
+		})
+		if err = tlsConn.Handshake(); err != nil {
+			return
+		}
+		defer tlsConn.Close()
+
+		conn = tlsConn
+	}
+
+	log.Debugf("creating SMTP client for %s ...", host)
+	client, err := smtp.NewClient(conn, host)
 	if err != nil {
-		panic(err)
+		return
 	}
+	defer client.Close()
 
-	var useTLS = false
-	if config["ssl"] != "" && strings.ToLower(config["ssl"]) == "true" {
-		useTLS = true
+	//swith to SSL/TLS
+	if !tls && starttls {
```

Comment:
seems the parm "starttls" is useless, you can alway check and fall back if it doesn't has the "STARTTLS" Extension?